app/src/main/kotlin/com/wire/android/config/CustomUiConfigurationProvider.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.config
+
+import androidx.compose.runtime.staticCompositionLocalOf
+import com.wire.android.BuildConfig
+
+/**
+ * A provider which provides flags for the custom build configuration related to the UI
+ *
+ */
+object CustomUiConfigurationProvider {
+    val isAccountCreationAllowed = BuildConfig.ALLOW_ACCOUNT_CREATION ?: true
+}
```

Comment:
How does it differ from the `FeatureVisibilityFlags` that we already have? Couldn't we just add this `isAccountCreationAllowed` there? 🤔

Reply:
It differs in a sense that this is coming from the config and that other one is just something we will ditch later in the future

Reply:
I was unaware of this other `FeatureVisibilityFlags`. But it does seem to be a "temporary" thing, as features are being developed. In any case, I guess it's indeed different from this one and can be left aside.

The overview is that:

Some flags might come from build time (the `default.json` or `custom.json` file).
Some might come from the backend (either backend settings or team settings).

Going forward, we should have a single place to handle UI flags and it shouldn't matter where they come from. Use the current user + Kalium's user scope + build flags and centralise everything into a single flag provider.

Reply:
We could not ditch that, for me it's the same and actually `FeatureVisibilityFlags` fits better this purpose, as we're enabling or disabling a feature of creating account 😄 
But ok, it's not that much important, just for me at least, having two very similar objects that are basically responsible for the same thing makes it a little bit more confusing 😉